src/main/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategyFactory.java

```diff
@@ -80,11 +84,27 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
      * @return true if running in the Kubernetes environment
      */
     @Override
-    @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
     public boolean isAutoDetectionApplicable() {
+        return tokenFileExists() && defaultKubernetesMasterReachable();
+    }
+
+    @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
+    boolean tokenFileExists() {
         return new File("/var/run/secrets/kubernetes.io/serviceaccount/token").exists();
     }
 
+    private boolean defaultKubernetesMasterReachable() {
+        try {
+            InetAddress.getByName("kubernetes.default.svc");
```

Comment:
what is a timeout for this call?

Reply:
actually, perhaps it's not all that relevant. it's guarded by the fs token check. 
So if someone is running a k8s cluster with slow DNS then it's OK if this is slow too.

Reply:
Yeah, it's only for corner cases, like you're running in Kubernetes, but kube-dns is not configured.